modules/secure-cloud-function-core/main.tf

```diff
@@ -20,14 +20,42 @@ data "google_project" "project" {
 
 module "cloudfunction_bucket" {
   source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
-  version = "~> 4.0"
+  version = "~>4.0"
 
   project_id    = var.project_id
   labels        = var.labels
   name          = "gcf-v2-sources-${data.google_project.project.number}-${var.location}"
   location      = var.location
   storage_class = "REGIONAL"
   force_destroy = var.force_destroy
+  cors = [
+    {
+      max_age_seconds = 0
+      method = [
+        "GET",
+      ]
+      origin = [
+        "https://*.cloud.google.com",
+        "https://*.corp.google.com",
+        "https://*.corp.google.com:*",
+        "https://*.cloud.google",
+        "https://*.byoid.goog",
+      ]
+      response_header = []
+    }
+  ]
+  lifecycle_rules = [{
+    action = {
+      type = "Delete"
+    }
+    condition = {
+      age                        = 0
+      days_since_custom_time     = 0
+      days_since_noncurrent_time = 0
+      num_newer_versions         = 3
+      with_state                 = "ARCHIVED"
+    }
+  }]
```

Comment:
Should any of this be parameterized, since it is part of the module? Or are these settings part of requirements for usage